packages/devtools_app/test/flutter/story_of_layout/arrow_test.dart

```diff
@@ -0,0 +1,92 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:devtools_app/src/inspector/flutter/story_of_your_layout/arrow.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  Widget wrap(Widget widget) => MaterialApp(home: widget);
+
+  group('Arrow Golden Tests', () {
+    group('Unidirectional', () {
+      Widget buildUnidirectionalArrowWrapper(ArrowType type) => wrap(
+            Container(
+              width: 100,
+              height: 100,
+              child: ArrowWrapper.unidirectional(
+                child: const Text('1'),
+                type: type,
+                arrowStrokeWidth: 2.0,
+                arrowColor: Colors.black,
+                arrowHeadSize: 8.0,
+              ),
+            ),
+          );
+      testWidgets('Unidirectional Left', (WidgetTester tester) async {
+        final widget = buildUnidirectionalArrowWrapper(ArrowType.left);
+        await tester.pumpWidget(widget);
+        await expectLater(
+          find.byWidget(widget),
+          matchesGoldenFile('goldens/arrow_unidirectional_left.png'),
+        );
+      });
+      testWidgets('Unidirectional Up', (WidgetTester tester) async {
+        final widget = buildUnidirectionalArrowWrapper(ArrowType.up);
+        await tester.pumpWidget(widget);
+        await expectLater(
+          find.byWidget(widget),
+          matchesGoldenFile('goldens/arrow_unidirectional_up.png'),
+        );
+      });
+      testWidgets('Unidirectional Right', (WidgetTester tester) async {
+        final widget = buildUnidirectionalArrowWrapper(ArrowType.right);
+        await tester.pumpWidget(widget);
+        await expectLater(
+          find.byWidget(widget),
+          matchesGoldenFile('goldens/arrow_unidirectional_right.png'),
+        );
+      });
+      testWidgets('Unidirectional Down', (WidgetTester tester) async {
+        final widget = buildUnidirectionalArrowWrapper(ArrowType.down);
+        await tester.pumpWidget(widget);
+        await expectLater(
+          find.byWidget(widget),
+          matchesGoldenFile('goldens/arrow_unidirectional_down.png'),
+        );
+      });
+    });
+    group('Bidirectional', () {
+      Widget buildBidirectionalArrowWrapper(Axis direction) => wrap(
+            Container(
+              width: 100,
+              height: 100,
+              child: ArrowWrapper.bidirectional(
+                child: const Text('1'),
+                direction: direction,
+                arrowStrokeWidth: 2.0,
+                arrowColor: Colors.black,
+                arrowHeadSize: 8.0,
+              ),
+            ),
+          );
+      testWidgets('Bidirectional Horizontal', (WidgetTester tester) async {
```

Comment:
nit: you can omit the word Bidirectional from these `testWidgets` call as the group already has the word `Bidirectional` in it.

In general these tests are good. There are no bad tests unless they are flaky and these test some useful cases to verify. :)

Reply:
Thanks Jacob, that's great to hear!

Unfortunately these golden tests fail on Flutter Web (chrome) and I don't know why ( have tried researching it as well :-( ) , this is the error message:
```
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following _Exception was thrown while running async test code:
Exception: Incorrect sequence of push/pop operations while building scene surfaces. After building
the scene the persisted surface stack must contain a single element which corresponds to the scene
itself (_PersistedScene). All other surfaces should have been popped off the stack. Found the
following surfaces in the stack:
PersistedScene, PersistedTransform
```

I will investigate it tomorrow!